hloc/colmap_from_opensfm.py

```diff
@@ -0,0 +1,171 @@
+import argparse
+import csv
+import json
+import math
+import typing
+from pathlib import Path
+
+import numpy as np
+from tqdm import tqdm
+
+from hloc import logger
+from hloc.utils.read_write_model import (
+    CAMERA_MODEL_NAMES,
+    Camera,
+    Image,
+    Point3D,
+    write_model,
+)
+
+
+def angle_axis_to_quaternion(angle_axis: np.ndarray) -> typing.List[float]:
+    angle = np.linalg.norm(angle_axis)
+    x = angle_axis[0] / angle
+    y = angle_axis[1] / angle
+    z = angle_axis[2] / angle
+    qw = math.cos(angle / 2.0)
+    qx = x * math.sqrt(1 - qw * qw)
+    qy = y * math.sqrt(1 - qw * qw)
+    qz = z * math.sqrt(1 - qw * qw)
+    return np.array([qw, float(qx), float(qy), float(qz)])
+
+
+def read_tracks(opensfm_path):
```

Comment:
future thinking, I think I've seen some implementation of reading tracks a few times, I believe there would be some value in a standalone opensfm_utils library we use internally

Reply:
That is a very good idea. In general, OpenSfM has all the utilities to read/write tracks/features/matches/reconstructions.
What we are missing is a standalone library, as you suggest, that will not have OpenSfM as a dependency and will be able to read/write in OpenSfM format.
I will try to gather the corresponding functions and make a small library. Created this ticket: https://dronedeploy.atlassian.net/browse/VIS-4663